File: euler_002.py
from time import time

def fib():
    prev, curr = 1, 2
    yield prev
    yield curr
    while True:
        [curr, prev] = [prev + curr, curr]
        yield curr

fib_gen = fib()

start = time()

fibonacci = next(fib_gen)
cumulated = 0

while fibonacci < 4000000:
    if fibonacci%2 == 0:
        cumulated += fibonacci
    fibonacci = next(fib_gen)

stop = time()

print("Sum is " + str(cumulated))
print("Time: " + str(stop-start))

File: euler_002.js
const start = Date.now();

let fib = [1,2,3];

let sum = 2;

while (fib[2] <= 4000000) {
    fib = [fib[1], fib[2], fib[1] + fib[2]];
    if (fib[2] % 2 === 0) {
        sum += fib[2];
    }
}


const end = Date.now();

console.log("Execute time: ", end - start, " Sum is: ", sum);